src/persist-client/src/usage.rs

```diff
@@ -205,61 +253,82 @@ impl StorageUsageClient {
     }
 
     async fn blob_raw_usage(&self, prefix: BlobKeyPrefix<'_>) -> BlobUsage {
-        retry_external(
-            &self.metrics.retries.external.storage_usage_shard_size,
-            || async {
-                let mut start = Instant::now();
-                let mut keys = 0;
-                let mut usage = BlobUsage::default();
-                self.blob
-                    .list_keys_and_metadata(&prefix.to_string(), &mut |metadata| {
-                        // Increment the step timing metrics as we go, so it
-                        // doesn't all show up at the end.
-                        keys += 1;
-                        if keys % 100 == 0 {
-                            let now = Instant::now();
-                            self.metrics
-                                .audit
-                                .step_blob_metadata
-                                .inc_by(now.duration_since(start).as_secs_f64());
-                            start = now;
-                        }
-
-                        match BlobKey::parse_ids(metadata.key) {
-                            Ok((shard, partial_blob_key)) => {
-                                let shard_usage = usage.by_shard.entry(shard).or_default();
+        let mut blob_usage_tasks = vec![];
+        for prefix in prefix.to_blob_prefixes() {
+            let blob = Arc::clone(&self.blob);
+            let metrics = Arc::clone(&self.metrics);
+            let name = format!("blob_raw_usage::prefix({})", prefix);
+            let task = mz_ore::task::spawn(|| name, async move {
```

Comment:
sorry to be a bother, but I don't love the structure of having this concurrency inside `blob_raw_usage`. how would you feel about instead:
- blob_raw_usage and the BlobKeyPrefix display impl stay as they were before
- we add a new `AllPrefix(String)` variant to `BlobKeyPrefix`
- some new method (`BlobKeyPrefix::all_prefixes()`?) returns the set of s0, s1, ... prefixes
- then we push the concurrency into `shards_usage` (potentially replacing the tasks with the same FuturesUnordered pattern used for shard_usage)

Reply:
oh yeah this code was the fastest beeline I could make to have a build I could try on staging (which seems to work / is a bit faster). definitely planning on fixing up the organization/concurrency